### PR TITLE
Add test cases to check maximum vCPUs with SE guest

### PR DESCRIPTION
**Pull request by fbq815**
Secure_execution: add test cases for both positive and negative to check
if SE guest could get 247 for maximum vCPU and the error message is
expected when it get 248 vCPUs

ID: 2209269

Signed-off-by: Boqiao Fu <bfu@redhat.com>

### CODE REVIEW

**Comment by fbq815** [comment]
Test log:
JOB ID     : 7a4a34b24acdba068129b486b918abece7f5dcbf
JOB LOG    : /root/avocado/job-results/job-2023-05-31T05.23-7a4a34b/job.log
 (1/2) Host_RHEL.m8.u9.product_rhel.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.9.0.s390x.io-github-autotest-qemu.secure_execution.247_max_vCPUs.s390-virtio: STARTED
 (1/2) Host_RHEL.m8.u9.product_rhel.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.9.0.s390x.io-github-autotest-qemu.secure_execution.247_max_vCPUs.s390-virtio: PASS (30.30 s)
 (2/2) Host_RHEL.m8.u9.product_rhel.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.9.0.s390x.io-github-autotest-qemu.secure_execution.248_vCPUs_error_check.s390-virtio: STARTED
 (2/2) Host_RHEL.m8.u9.product_rhel.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.9.0.s390x.io-github-autotest-qemu.secure_execution.248_vCPUs_error_check.s390-virtio: PASS (5.77 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2023-05-31T05.23-7a4a34b/results.html
JOB TIME   : 37.71 s

@PaulYuuu @zixi-chen please help me review this patch

File: qemu/tests/secure_execution.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    1) Log into a secure guest
+    2) Check if there's error messagein
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    timeout = float(params.get("login_timeout", 240))
+    serial_login = params.get("serial_login", "no") == "yes"
```
Comment (PaulYuuu) [inline qemu/tests/secure_execution.py:18]: Can use `params.get_boolean`

File: qemu/tests/secure_execution.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    Secured guest boot test:
+    1) Log into a secure guest
+    2) Check if there's error messagein
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    timeout = float(params.get("login_timeout", 240))
```
Comment (PaulYuuu) [inline qemu/tests/secure_execution.py:17]:
```suggestion
```

File: qemu/tests/secure_execution.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    serial_login = params.get("serial_login", "no") == "yes"
+    vm = env.get_vm(params["main_vm"])
+    error_context.context("Try to log into guest '%s'." % vm.name,
+                          test.log.info)
+    error_check = params.get("error_check", "no")
+    if params.get("start_vm") == "no" and "no" in error_check:
```
Comment (PaulYuuu) [inline qemu/tests/secure_execution.py:23]: Same above, use get_boolean to check if we need error_check.

Comment (fbq815) [reply qemu/tests/secure_execution.py:23]: the error_check here is a param in cfg file, separate valid and invalid scenarios

File: qemu/tests/secure_execution.py (hunk 1 above)
Comment (fbq815) [reply qemu/tests/secure_execution.py:18]: this is not a check point, it's more like a parameter leave for future features, once other test case is not able to use serial_login

File: qemu/tests/secure_execution.py (hunk 2 above)
Comment (fbq815) [reply qemu/tests/secure_execution.py:17]: not get the point, why do I need to delete the timeout here, is it find if I change it to timeout = params.get_numeric("login_timeout", 360)?

File: qemu/tests/secure_execution.py (hunk 1 above)
Comment (PaulYuuu) [reply qemu/tests/secure_execution.py:18]:
```python
    def get_boolean(self, key, default=False):
        """
        Check if a config option is set to a default affirmation or not.

        :param key: config option key
        :type key: str
        :param bool default: whether to assume "yes" or "no" if param
                             does not exist (defaults to False/"no").
        :return whether option is set to 'yes'
        :rtype: bool
        """
        value = self.get(key, "yes" if default else "no")
        if value in ("yes", "on", "true"):
            return True
        if value in ("no", "off", "false"):
            return False
        raise ValueError("Cannot get boolean parameter value for %s: %s", key, value)
```

File: qemu/tests/secure_execution.py (hunk 2 above)
Comment (fbq815) [reply qemu/tests/secure_execution.py]: removed

File: qemu/tests/secure_execution.py (hunk 1 above)
Comment (fbq815) [reply qemu/tests/secure_execution.py]: removed

File: qemu/tests/secure_execution.py (hunk 3 above)
Comment (fbq815) [reply qemu/tests/secure_execution.py]: updated

**Comment by fbq815** [comment]
Test result:
JOB ID     : 3b4bc955ae8b4d57b5c1d2a6161ae6a1553b9458
JOB LOG    : /root/avocado/job-results/job-2023-07-06T03.02-3b4bc95/job.log
 (1/4) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.s390-virtio: STARTED
 (1/4) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.s390-virtio: PASS (424.01 s)
 (2/4) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.secure_img.s390-virtio: STARTED
 (2/4) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.secure_img.s390-virtio: PASS (38.96 s)
 (3/4) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.secure_execution.247_max_vCPUs.s390-virtio: STARTED
 (3/4) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.secure_execution.247_max_vCPUs.s390-virtio: PASS (49.53 s)
 (4/4) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.secure_execution.248_vCPUs_error_check.s390-virtio: STARTED
 (4/4) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.secure_execution.248_vCPUs_error_check.s390-virtio: PASS (5.41 s)
RESULTS    : PASS 4 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2023-07-06T03.02-3b4bc95/results.html
JOB TIME   : 520.64 s

**Comment by fbq815** [comment]
@PaulYuuu please help me review this patch

File: qemu/tests/secure_execution.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        vm.destroy()
+    else:
+        error_msg = params.get("error_msg", "")
+        try:
+            vm.create(params=params)
+            vm.verify_alive()
+            output = vm.process.get_output()
+            vm.destroy()
+        except virt_vm.VMCreateError as detail:
+            output = str(detail)
```
Comment (PaulYuuu) [inline qemu/tests/secure_execution.py:34]: I would suggest `output = detail.output`

File: qemu/tests/secure_execution.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    vm = env.get_vm(params["main_vm"])
+    error_context.context("Try to log into guest '%s'." % vm.name,
+                          test.log.info)
+    if params.get("start_vm") == "no" and not params.get_boolean("error_check"):
```
Comment (PaulYuuu) [inline qemu/tests/secure_execution.py:20]: Seems `start_vm` is always to `no` in this case.

File: qemu/tests/secure_execution.py (hunk 4 above)
Comment (fbq815) [reply qemu/tests/secure_execution.py:34]: updated

File: qemu/tests/secure_execution.py (hunk 5 above)
Comment (fbq815) [reply qemu/tests/secure_execution.py:20]: this is for future cases in this cfg

**Comment by fbq815** [comment]
Test result:
JOB ID     : c2e7d7197dee38c694b801742ae3be7e6b03ff99
JOB LOG    : /root/avocado/job-results/job-2023-07-11T02.55-c2e7d71/job.log
 (1/2) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.secure_execution.247_max_vCPUs.s390-virtio: STARTED
 (1/2) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.secure_execution.247_max_vCPUs.s390-virtio: PASS (29.16 s)
 (2/2) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.secure_execution.248_vCPUs_error_check.s390-virtio: STARTED
 (2/2) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.secure_execution.248_vCPUs_error_check.s390-virtio: PASS (5.51 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2023-07-11T02.55-c2e7d71/results.html
JOB TIME   : 36.42 s

**Comment by fbq815** [comment]
Test result:
JOB ID     : 40c1193478d5127163b9b2c0d3fb186a4c5984f0
JOB LOG    : /root/avocado/job-results/job-2023-07-11T03.46-40c1193/job.log
 (1/2) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.secure_execution.247_max_vCPUs.s390-virtio: STARTED
 (1/2) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.secure_execution.247_max_vCPUs.s390-virtio: PASS (28.28 s)
 (2/2) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.secure_execution.248_vCPUs_error_check.s390-virtio: STARTED
 (2/2) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.secure_execution.248_vCPUs_error_check.s390-virtio: PASS (5.17 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2023-07-11T03.46-40c1193/results.html
JOB TIME   : 35.23 s

File: qemu/tests/secure_execution.py (hunk 5 above)
Comment (fbq815) [reply qemu/tests/secure_execution.py]: updated

**Review by PaulYuuu — APPROVED** [review]
